indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceRecordReader.java

```diff
@@ -70,18 +69,18 @@ public void initialize(InputSplit split, final TaskAttemptContext context) throw
   {
     spec = readAndVerifyDatasourceIngestionSpec(context.getConfiguration(), HadoopDruidIndexerConfig.jsonMapper);
 
-    List<DataSegment> segments = ((DatasourceInputSplit) split).getSegments();
+    List<WindowedDataSegment> segments = ((DatasourceInputSplit) split).getSegments();
 
-    List<StorageAdapter> adapters = Lists.transform(
+    List<WindowedStorageAdapter> adapters = Lists.transform(
         segments,
-        new Function<DataSegment, StorageAdapter>()
+        new Function<WindowedDataSegment, WindowedStorageAdapter>()
         {
           @Override
-          public StorageAdapter apply(DataSegment segment)
+          public WindowedStorageAdapter apply(WindowedDataSegment segment)
           {
             try {
-              logger.info("Getting storage path for segment [%s]", segment.getIdentifier());
-              Path path = new Path(JobHelper.getURIFromSegment(segment));
+              logger.info("Getting storage path for segment [%s]", segment.getSegment().getIdentifier());
+              Path path = new Path(JobHelper.getURIFromSegment(segment.getSegment()));
```

Comment:
does this file get used anywhere outside of Hadoop? If yes, this will cause dependency woes